File: lab-2.3/c#/Hulk.cs
namespace CharacterLib
{
    public class Hulk : IDescribable
    {
        private string name;
        private int strengthLevel;

        public Hulk()
        {
            name = "Unknown";
            strengthLevel = 80;
        }

        public Hulk(string name, int strengthLevel)
        {
            this.name = name;
            this.strengthLevel = strengthLevel;
        }

        public Hulk(Hulk other)
        {
            name = other.name;
            strengthLevel = other.strengthLevel;
        }

        public string GetName()
        {
            return name;
        }

        public int GetPower()
        {
            return strengthLevel;
        }

        public string Describe()
        {
            return $"Hulk: {name}, Strength: {strengthLevel}";
        }
    }
}

File: lab-2.3/c++/Hulk.cpp
#include "Hulk.h"

Hulk::Hulk() : name("Unknown"), strengthLevel(80) {}

Hulk::Hulk(std::string n, int sl)
    : name(n), strengthLevel(sl) {}

Hulk::Hulk(const Hulk& other)
    : name(other.name), strengthLevel(other.strengthLevel) {}

std::string Hulk::getName() const {
    return name;
}

int Hulk::getPower() const {
    return strengthLevel;
}

std::string Hulk::describe() const {
    return "Hulk: " + name + ", Strength: " + std::to_string(strengthLevel);
}

IDescribable* Hulk::clone() const {
    return new Hulk(*this);
}